nautobot_golden_config/views.py

```diff
@@ -824,41 +825,47 @@ class ComplianceRuleBulkEditView(generic.BulkEditView):
 #
 # GoldenConfigSetting
 #
+class GoldenConfigSettingView(generic.ObjectView):
+    """View for single GoldenConfigSetting instance."""
 
+    queryset = models.GoldenConfigSetting.objects.all()
 
-class GoldenConfigSettingView(generic.ObjectView):
-    """View for the only GoldenConfigSetting instance."""
+    # def get_extra_context(self, request, instance):
+    #     """Add extra data to detail view for Nautobot."""
+    #     return {}
+
+
+class GoldenConfigSettingCreateView(generic.ObjectEditView):
+    """Create view."""
 
+    model = models.GoldenConfigSetting
     queryset = models.GoldenConfigSetting.objects.all()
+    model_form = forms.GoldenConfigSettingFeatureForm
 
-    def get(self, request, *args, **kwargs):
-        """Override the get parameter to get the first instance to enforce singleton pattern."""
-        instance = self.queryset.first()
 
-        return render(
-            request,
-            self.get_template_name(),
-            {
-                "object": instance,
-                **self.get_extra_context(request, instance),
-            },
-        )
+class GoldenConfigSettingDeleteView(generic.ObjectDeleteView):
+    """Delete view."""
 
-    def get_extra_context(self, request, instance):
-        """Add extra data to detail view for Nautobot."""
-        return {}
+    model = models.GoldenConfigSetting
+    queryset = models.GoldenConfigSetting.objects.all()
 
 
 class GoldenConfigSettingEditView(generic.ObjectEditView):
-    """View for editing the Global configurations."""
+    """Edit view."""
 
+    model = models.GoldenConfigSetting
     queryset = models.GoldenConfigSetting.objects.all()
     model_form = forms.GoldenConfigSettingFeatureForm
-    default_return_url = "plugins:nautobot_golden_config:goldenconfigsetting"
+    template_name = "nautobot_golden_config/goldenconfigsetting_edit.html"
```

Comment:
why do we need a custom edit template?

Reply:
sorry for asking again, but I still don't get the point to handle the 3 as separate sections. is it just for UX convenience?